comment/models/flags.py

```diff
@@ -27,7 +27,12 @@ class Flag(models.Model):
     comment = models.OneToOneField(Comment, on_delete=models.CASCADE)
     count = models.PositiveIntegerField(default=0)
     state = models.SmallIntegerField(choices=STATES_CHOICES, default=UNFLAGGED)
-    moderator = models.ForeignKey(User, on_delete=models.SET_NULL, null=True, blank=True)
+    moderator = models.ForeignKey(
```

Comment:
Do we need moderator field for the flag anymore?

Reply:
yes, it is used to store information when a user toggles the flag state. See the code here : https://github.com/Radi85/Comment/blob/master/comment/models/flags.py#L82

Also, now that you have raised the point, what do you think about adding a soft delete option that preserves the comment when a `comment_moderator` deletes a comment. When `comment_admins` deletes a comment, I think it could be permanently deleted.

Reply:
> Also, now that you have raised the point, what do you think about adding a soft delete option that preserves the comment when a `comment_moderator` deletes a comment. When `comment_admins` deletes a comment, I think it could be permanently deleted.

Agree. That would be nice.